langchain/src/vectorstores/mongodb_atlas.ts

```diff
@@ -53,14 +58,24 @@ export class MongoDBAtlasVectorSearch extends VectorStore {
   async similaritySearchVectorWithScore(
     query: number[],
     k: number,
-    preFilter?: MongoDBDocument,
-    postFilterPipeline?: MongoDBDocument[]
+    filter?: MongoDBAtlasFilter
   ): Promise<[Document, number][]> {
     const knnBeta: MongoDBDocument = {
       vector: query,
       path: this.embeddingKey,
       k,
     };
```

Comment:
@jacoblee93, rows 68-77: this code is not working in case we have only a postFilterPipeline and no preFilter.
My proposal:

 ```
    let preFilter: MongoDBDocument | undefined;
    let postFilterPipeline: MongoDBDocument[] | undefined;
    if (filter?.preFilter || filter?. postFilterPipeline) {
        preFilter = filter.preFilter;
        postFilterPipeline = filter.postFilterPipeline;
    } else 
      filter &&  (preFilter = filter);
```

Reply:
ok pushed my proposal if your checks are good LGTM overall for me and you can release thanks

Reply:
Ah duh, thank you. Will get this out today!